### DISCUSSION TITLE
Global setup

### DISCUSSION THREAD

**Discussion by borjalo** [gladly-team/next-firebase-auth#525] [Q&A]
I was implementing this solution in my Next.js app and wondered if there's a way to set the functions `withAuthUser`, `withAuthUserSSR`, etc in a global way instead of setting them in each page. I tried to put them in my `_app.tsx` file but once I reload the page it's showing me this error in the `login.tsx` when trying to do a `const auth = getAuth()` from the Firebase JS SDK:

> FirebaseError: Firebase: No Firebase App '[DEFAULT]' has been created - call Firebase App.initializeApp() (app/no-app).

<img width="958" alt="image" src="https://user-images.githubusercontent.com/22989690/184102841-325d6524-fd78-40d4-b2c8-08b581372f0e.png">

I guess there's no need to initialize the Firebase app again as it's already initialized by `next-firebase-auth`

Any ideas?

**Comment by kmjennison** [comment]
For setting an auth requirement globally for the app, there are a few options:
1. Create a common HOC with shared logic and use it on each page
2. Create a "layout" component that uses `useAuthUser` and is rendered in _app.js
3. Use middleware, which is not yet supported but tracked here: https://github.com/gladly-team/next-firebase-auth/issues/418

Related discussions with more info:
* https://github.com/gladly-team/next-firebase-auth/issues/64#issuecomment-776014927
* https://github.com/gladly-team/next-firebase-auth/issues/155#issuecomment-912809559
